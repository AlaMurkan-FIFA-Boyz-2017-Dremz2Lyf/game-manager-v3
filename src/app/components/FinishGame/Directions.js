import React from 'react';
import { Segment, Header, Icon, Message } from 'semantic-ui-react';

const Directions = () => (
  <Segment basic textAlign="center">
    <Header color="teal" as="h2">
      <Icon name="share" />
      <Header.Content>
        Directions
      </Header.Content>
    </Header>
    <Message color="teal">
      Please select Match Facts in the post-match menu. When there, please use the share feature to take a screenshot.
    </Message>
  </Segment>
);

export default Directions;
