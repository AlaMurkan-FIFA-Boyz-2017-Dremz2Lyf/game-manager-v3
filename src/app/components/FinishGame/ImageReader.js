import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Segment, Progress, Message } from 'semantic-ui-react';

import Tesseract from 'tesseract.js';

const whitelist = 'AYRSTPOTFIC%0123456789hotsnfargjetiwckluyd ';

class ImageReader extends PureComponent {
  static propTypes = {
    images: PropTypes.array,
  }
  state = {
    statsPercent: 0,
    scorePercent: 0,
    results: null,
  }

  tesseractJob = Tesseract.create({
    workerPath: 'https://cdn.rawgit.com/naptha/tesseract.js/1.0.10/dist/worker.js',
    langPath: 'https://cdn.rawgit.com/naptha/tessdata/gh-pages/3.02/',
  });

  errorMessage = 'Welp, something went wrong. Tell Scott to fix his crap.'

  readImages = (images) => {
    const promises = images.map((image) => {
      console.log({ image });
      return Promise.resolve(this.tesseractJob.recognize(image, {
        lang: 'eng',
        tessedit_char_whitelist: whitelist,
      }).progress(progress => console.log({ progress })).catch((error) => {
        this.setState({ error });
      }));
    });

    return Promise.all(promises).then((results) => {
      const final = results.map(({ lines }) => lines.map(({ text }) => text));
      this.tesseractJob.terminate();
      return final;
    }).then((results) => {
      this.setState({
        results,
      });
    });
  }

  render() {
    const { statsPercent, scorePercent, results, error } = this.state;
    const { images } = this.props;
    if (images.length === 2 && !results) {
      this.readImages(images);
    }

    return (
      <Segment basic>
        <Message color="teal">
          <Message.Header>
            Processing Image
          </Message.Header>
        </Message>
        {
          error ? (
            <Message color="red" header="Oops!" content={this.errorMessage} />
          ) : (
            <Segment basic>
              <p>{results}</p>
              <Progress percent={scorePercent}>
                Processing Score
              </Progress>
              <Progress percent={statsPercent}>
                Processing Stats
              </Progress>
            </Segment>
          )
        }
      </Segment>
    );
  }
}

export default ImageReader;
