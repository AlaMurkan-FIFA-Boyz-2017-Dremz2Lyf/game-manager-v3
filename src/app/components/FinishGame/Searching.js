import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Segment, Message, Loader } from 'semantic-ui-react';

class Searching extends PureComponent {
  static propTypes = {
    loading: PropTypes.bool,
    fetchImages: PropTypes.func,
  };

  componentDidMount() {
    this.props.fetchImages();
  }

  render() {
    const { loading } = this.props;

    return (
      <Segment basic>
        <Message color="teal">
          <Message.Header>
            Searching Twitter for your results.
          </Message.Header>
        </Message>
        {loading ? <Loader size="huge" active={loading} inline="centered" /> : null}
      </Segment>
    );
  }
}

export default Searching;
