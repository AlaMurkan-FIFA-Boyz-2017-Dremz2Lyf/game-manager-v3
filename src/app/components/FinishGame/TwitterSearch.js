import React from 'react';
import PropTypes from 'prop-types';
import { Segment, Message, Header, Icon } from 'semantic-ui-react';

const TwitterSearch = ({ searchTerm }) => (
  <Segment basic>
    <Header color="teal" as="h2">
      <Icon name="twitter" />
      <Header.Content>
        Use the Hashtag below to tweet the screenshot.
      </Header.Content>
    </Header>
    <Message color="teal">
      <Message.Header>
        {searchTerm}
      </Message.Header>
      <p>Hit next after the tweet has been sent.</p>
    </Message>
  </Segment>
);

TwitterSearch.propTypes = {
  searchTerm: PropTypes.string,
  generateHashtag: PropTypes.func,
};

export default TwitterSearch;
