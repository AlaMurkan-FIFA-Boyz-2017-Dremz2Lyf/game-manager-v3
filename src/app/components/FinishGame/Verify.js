import React from 'react';
import PropTypes from 'prop-types';
import { Header, Segment, Table } from 'semantic-ui-react';

const TRACKED_STATS = [
  'score',
  'shots',
  'shots on target',
  'possession %',
  'tackles',
  'fouls',
  'yellow cards',
  'red cards',
  'injuries',
  'offsides',
  'corners',
  'shot accuracy %',
  'pass accuracy %',
];

const Verify = ({ results }) => (
  <Segment basic>
    <Header color="teal">
      Verify
    </Header>
    <p>Please verify that the results match the image below.</p>
    <Table compact inverted unstackable>
      <Table.Header>
        <Table.Row>
          <Table.HeaderCell textAlign="left">Home</Table.HeaderCell>
          <Table.HeaderCell textAlign="center">Stat</Table.HeaderCell>
          <Table.HeaderCell textAlign="right">Away</Table.HeaderCell>
        </Table.Row>
      </Table.Header>

      <Table.Body>
        {TRACKED_STATS.map((statName) => {
          const home = results.home[statName];
          const away = results.away[statName];
          return (
            <Table.Row key={statName}>
              <Table.Cell textAlign="left">{home}</Table.Cell>
              <Table.Cell textAlign="center">{statName}</Table.Cell>
              <Table.Cell textAlign="right">{away}</Table.Cell>
            </Table.Row>
          );
        })}
      </Table.Body>
    </Table>
  </Segment>
);

Verify.propTypes = {
  results: PropTypes.object,
};

export default Verify;
