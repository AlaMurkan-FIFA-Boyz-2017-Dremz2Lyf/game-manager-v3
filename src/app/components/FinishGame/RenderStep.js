import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Header, Segment, Table } from 'semantic-ui-react';
import ImageReader from './ImageReader';
import Directions from './Directions';
import TwitterSearch from './TwitterSearch';
import Searching from './Searching';

const TRACKED_STATS = [
  'score',
  'shots',
  'shots on target',
  'possession %',
  'tackles',
  'fouls',
  'yellow cards',
  'red cards',
  'injuries',
  'offsides',
  'corners',
  'shot accuracy %',
  'pass accuracy %',
];

const Verify = ({ results }) => (
  <Segment basic>
    <Header color="teal">
      Verify
    </Header>
    <p>Please verify that the results match the image below.</p>
    <Table compact inverted unstackable>
      <Table.Header>
        <Table.Row>
          <Table.HeaderCell textAlign="left">Home</Table.HeaderCell>
          <Table.HeaderCell textAlign="center">Stat</Table.HeaderCell>
          <Table.HeaderCell textAlign="right">Away</Table.HeaderCell>
        </Table.Row>
      </Table.Header>

      <Table.Body>
        {TRACKED_STATS.map((statName) => {
          const home = results.home[statName];
          const away = results.away[statName];
          return (
            <Table.Row key={statName}>
              <Table.Cell textAlign="left">{home}</Table.Cell>
              <Table.Cell textAlign="center">{statName}</Table.Cell>
              <Table.Cell textAlign="right">{away}</Table.Cell>
            </Table.Row>
          );
        })}
      </Table.Body>
    </Table>
  </Segment>
);

Verify.propTypes = {
  results: PropTypes.object,
};


class RenderStep extends PureComponent {
  static propTypes = {
    step: PropTypes.number,
  }

  getStepComponent = (step) => {
    const stepMap = [
      Directions,
      TwitterSearch,
      Searching,
      ImageReader,
      Verify,
    ];

    return stepMap[step];
  }

  render() {
    const { step } = this.props;

    const CurrentStep = this.getStepComponent(step);
    return (
      <CurrentStep {...this.props} />
    );
  }
}

export default RenderStep;
