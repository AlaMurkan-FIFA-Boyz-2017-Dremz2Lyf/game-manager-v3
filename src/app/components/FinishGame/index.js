import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Segment, Responsive, Image, Header, Grid, Step, Icon, Button, Modal } from 'semantic-ui-react';
// import randomID from 'random-id';
import RenderStep from './RenderStep';

class FinishGame extends PureComponent {
  state = {
    loading: false,
    hashTag: '',
    step: 0,
    images: null,
    imageSource: '',
  }

  componentDidMount() {
    this.generateHashtag();
  }

  onSearchChange = (_, { hashTag }) => {
    this.setState({
      hashTag,
    });
  }

  onResults = (results) => {
    this.setState({
      results,
      loading: false,
      step: this.state.step + 1,
    });
  }

  HASHTAG_LENGTH = 10;
  baseUrl = process.env.LAMBDA_URL;

  fetchImages = () => {
    const { hashTag } = this.state;

    this.setState({ loading: true });

    fetch(`${this.baseUrl}${hashTag}`)
      .then(response => response.json())
      .then((results) => {
        const { images, imageSource } = results;

        this.setState({
          imageSource,
          images,
        }, () => {
          this.nextStep();
        });
      });
  }

  nextStep = () => {
    const nextStep = this.state.step + 1;
    if (nextStep < 5) {
      this.setState({
        step: nextStep,
      });
    } else {
      // TODO: submit mutation here.
      this.closeModal();
    }
  }

  closeModal = () => {
    const { closeModal } = this.props;
    closeModal();
  }

  generateHashtag = () => {
    // NOTE: for testing we use a locked tag. add this back in when ready
    // const hashTag = randomID(HASHTAG_LENGTH);
    const hashTag = '656liv4manure0';
    this.setState({
      hashTag,
    });
  }

  renderButtons = () => {
    const { step } = this.state;
    const disabled = step === 2 || step === 3;
    return (
      <Modal.Actions>
        <Button basic color="red" inverted onClick={this.closeModal}>
          <Icon name="remove" /> Cancel
        </Button>
        <Button disabled={disabled} color="green" inverted onClick={this.nextStep}>
          {step === 4 ? <Icon name="checkmark" /> : null}
          {step === 4 ? 'Verify' : 'Next'}
        </Button>
      </Modal.Actions>
    );
  }

  render() {
    const { step, imageSource, hashTag, loading, images, results } = this.state;

    const stepProps = {
      results,
      step,
      imageSource,
      hashTag,
      loading,
      images,
      fetchImages: this.fetchImages,
      onResults: this.onResults,
    };

    return (
      <Segment basic>
        <Modal.Content>
          <Grid columns={2}>
            <Grid.Row>
              <Grid.Column verticalAlign="middle" mobile={4} computer={6}>
                <Step.Group vertical size="mini">
                  <Step
                    completed={step > 0}
                    active={step === 0}
                  >
                    <Icon size="small" name="info" />
                    <Responsive as={Step.Content} {...Responsive.onlyComputer}>
                      <Step.Title>Directions</Step.Title>
                    </Responsive>
                  </Step>
                  <Step
                    completed={step > 1}
                    active={step === 1}
                  >
                    <Icon size="small" name="hashtag" />
                    <Responsive as={Step.Content} {...Responsive.onlyComputer}>
                      <Step.Title>Hashtag</Step.Title>
                    </Responsive>
                  </Step>
                  <Step
                    completed={step > 2}
                    active={step === 2}
                  >
                    <Icon size="small" name="twitter" />
                    <Responsive as={Step.Content} {...Responsive.onlyComputer}>
                      <Step.Title>Searching</Step.Title>
                    </Responsive>
                  </Step>
                  <Step
                    completed={step > 3}
                    active={step === 3}
                  >
                    <Icon size="small" name="spinner" loading={step ===3} />
                    <Responsive as={Step.Content} {...Responsive.onlyComputer}>
                      <Step.Title>Processing</Step.Title>
                    </Responsive>
                  </Step>
                  <Step
                    completed={step > 4}
                    active={step === 4}
                  >
                    <Icon size="small" name="pointing down" />
                    <Responsive as={Step.Content} {...Responsive.onlyComputer}>
                      <Step.Title>Confirm</Step.Title>
                    </Responsive>
                  </Step>
                </Step.Group>
              </Grid.Column>
              <Grid.Column mobile={12} computer={10}>
                <RenderStep {...stepProps} />
              </Grid.Column>
            </Grid.Row>
          </Grid>
        </Modal.Content>
        <Segment basic floated="right">
          {this.renderButtons()}
        </Segment>
        <Header color="teal" content="Results" />
        <Image
          src={imageSource}
          fluid
        />
      </Segment>
    );
  }
}
FinishGame.propTypes = {
  tournament: PropTypes.bool,
  closeModal: PropTypes.func,
};

export default FinishGame;
