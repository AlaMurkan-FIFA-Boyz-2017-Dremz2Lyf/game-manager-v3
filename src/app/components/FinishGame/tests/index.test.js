import { shallow } from 'enzyme';
import React from 'react';

import FinishGame from '../index';
import { combinedResults } from './tesseractMock';

describe('<FinishGame />', () => {
  let wrapper;
  let closeModalMock;
  let imageFetcherMock;

  beforeEach(() => {
    closeModalMock = jest.fn();
    imageFetcherMock = jest.fn((baseUrl, hashtag) => Promise.resolve({
      images: ['https://fake.url.com/image1', 'https://fake.url.com/image2'],
      imageSource: 'https://fake.url.com',
    }));
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should match snapshot and start with step 0', () => {
    wrapper = shallow(<FinishGame closeModal={closeModalMock} />);
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.state('step')).toBe(0);
  });

  describe('onResults - method', () => {
    it('should set results to state and increment the step counter', () => {
      wrapper = shallow(<FinishGame closeModal={closeModalMock} />);
      expect(wrapper.state('results')).toBeUndefined();
      expect(wrapper.state('step')).toBe(0);
      wrapper.instance().onResults(combinedResults);
      expect(wrapper.state('results')).toEqual(combinedResults);
      expect(wrapper.state('step')).toBe(1);
    });
  });

  describe('fetchImages', () => {
    it('should call the imageFetcher with the baseUrl and hashtag', () => {
      wrapper = shallow(
        <FinishGame closeModal={closeModalMock} imageFetcher={imageFetcherMock} />,
      );
      return wrapper.instance().fetchImages().then(() => {
        expect(imageFetcherMock).toHaveBeenCalled();
        expect(imageFetcherMock).toHaveBeenCalledWith('https://fake.url.com', '656liv4manure0');
      });
    });
  });
});
