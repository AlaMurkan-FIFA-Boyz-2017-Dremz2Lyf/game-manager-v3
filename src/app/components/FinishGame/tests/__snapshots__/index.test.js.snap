// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FinishGame /> should match snapshot and start with step 0 1`] = `
<Segment
  basic={true}
>
  <ModalContent>
    <Grid
      columns={2}
    >
      <GridRow>
        <GridColumn
          computer={6}
          mobile={4}
          verticalAlign="middle"
        >
          <StepGroup
            size="mini"
            vertical={true}
          >
            <Step
              active={true}
              completed={false}
            >
              <Icon
                as="i"
                name="info"
                size="small"
              />
              <Responsive
                as={[Function]}
                minWidth={992}
              >
                <StepTitle>
                  Directions
                </StepTitle>
              </Responsive>
            </Step>
            <Step
              active={false}
              completed={false}
            >
              <Icon
                as="i"
                name="hashtag"
                size="small"
              />
              <Responsive
                as={[Function]}
                minWidth={992}
              >
                <StepTitle>
                  Hashtag
                </StepTitle>
              </Responsive>
            </Step>
            <Step
              active={false}
              completed={false}
            >
              <Icon
                as="i"
                name="twitter"
                size="small"
              />
              <Responsive
                as={[Function]}
                minWidth={992}
              >
                <StepTitle>
                  Searching
                </StepTitle>
              </Responsive>
            </Step>
            <Step
              active={false}
              completed={false}
            >
              <Icon
                as="i"
                loading={false}
                name="spinner"
                size="small"
              />
              <Responsive
                as={[Function]}
                minWidth={992}
              >
                <StepTitle>
                  Processing
                </StepTitle>
              </Responsive>
            </Step>
            <Step
              active={false}
              completed={false}
            >
              <Icon
                as="i"
                name="pointing down"
                size="small"
              />
              <Responsive
                as={[Function]}
                minWidth={992}
              >
                <StepTitle>
                  Confirm
                </StepTitle>
              </Responsive>
            </Step>
          </StepGroup>
        </GridColumn>
        <GridColumn
          computer={10}
          mobile={12}
        >
          <RenderStep
            fetchImages={[Function]}
            hashtag="656liv4manure0"
            imageSource=""
            images={null}
            loading={false}
            onResults={[Function]}
            processer={
              TesseractWorker {
                "_currentJob": null,
                "_queue": Array [],
                "create": [Function],
                "version": "1.0.10",
                "worker": null,
                "workerOptions": Object {
                  "langPath": "http://cdn.rawgit.com/naptha/tessdata/gh-pages/3.02/",
                  "workerPath": "/Users/scotthorn/repos/Game-Manager-V3/node_modules/tesseract.js/src/node/worker.js",
                },
              }
            }
            step={0}
          />
        </GridColumn>
      </GridRow>
    </Grid>
  </ModalContent>
  <Segment
    basic={true}
    floated="right"
  >
    <ModalActions>
      <Button
        as="button"
        basic={true}
        color="red"
        inverted={true}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="remove"
        />
         Cancel
      </Button>
      <Button
        as="button"
        color="green"
        disabled={false}
        inverted={true}
        onClick={[Function]}
      >
        Next
      </Button>
    </ModalActions>
  </Segment>
  <Header
    color="teal"
    content="Results"
  />
  <Image
    as="img"
    fluid={true}
    src=""
    ui={true}
  />
</Segment>
`;

exports[`<FinishGame /> should render a checkmark Icon and the word "Verify" if the current step is 4 1`] = `
<Segment
  basic={true}
>
  <ModalContent>
    <Grid
      columns={2}
    >
      <GridRow>
        <GridColumn
          computer={6}
          mobile={4}
          verticalAlign="middle"
        >
          <StepGroup
            size="mini"
            vertical={true}
          >
            <Step
              active={true}
              completed={false}
            >
              <Icon
                as="i"
                name="info"
                size="small"
              />
              <Responsive
                as={[Function]}
                minWidth={992}
              >
                <StepTitle>
                  Directions
                </StepTitle>
              </Responsive>
            </Step>
            <Step
              active={false}
              completed={false}
            >
              <Icon
                as="i"
                name="hashtag"
                size="small"
              />
              <Responsive
                as={[Function]}
                minWidth={992}
              >
                <StepTitle>
                  Hashtag
                </StepTitle>
              </Responsive>
            </Step>
            <Step
              active={false}
              completed={false}
            >
              <Icon
                as="i"
                name="twitter"
                size="small"
              />
              <Responsive
                as={[Function]}
                minWidth={992}
              >
                <StepTitle>
                  Searching
                </StepTitle>
              </Responsive>
            </Step>
            <Step
              active={false}
              completed={false}
            >
              <Icon
                as="i"
                loading={false}
                name="spinner"
                size="small"
              />
              <Responsive
                as={[Function]}
                minWidth={992}
              >
                <StepTitle>
                  Processing
                </StepTitle>
              </Responsive>
            </Step>
            <Step
              active={false}
              completed={false}
            >
              <Icon
                as="i"
                name="pointing down"
                size="small"
              />
              <Responsive
                as={[Function]}
                minWidth={992}
              >
                <StepTitle>
                  Confirm
                </StepTitle>
              </Responsive>
            </Step>
          </StepGroup>
        </GridColumn>
        <GridColumn
          computer={10}
          mobile={12}
        >
          <RenderStep
            fetchImages={[Function]}
            hashtag="656liv4manure0"
            imageSource=""
            images={null}
            loading={false}
            onResults={[Function]}
            processer={
              TesseractWorker {
                "_currentJob": null,
                "_queue": Array [],
                "create": [Function],
                "version": "1.0.10",
                "worker": null,
                "workerOptions": Object {
                  "langPath": "http://cdn.rawgit.com/naptha/tessdata/gh-pages/3.02/",
                  "workerPath": "/Users/scotthorn/repos/Game-Manager-V3/node_modules/tesseract.js/src/node/worker.js",
                },
              }
            }
            step={0}
          />
        </GridColumn>
      </GridRow>
    </Grid>
  </ModalContent>
  <Segment
    basic={true}
    floated="right"
  >
    <ModalActions>
      <Button
        as="button"
        basic={true}
        color="red"
        inverted={true}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="remove"
        />
         Cancel
      </Button>
      <Button
        as="button"
        color="green"
        disabled={false}
        inverted={true}
        onClick={[Function]}
      >
        Next
      </Button>
    </ModalActions>
  </Segment>
  <Header
    color="teal"
    content="Results"
  />
  <Image
    as="img"
    fluid={true}
    src=""
    ui={true}
  />
</Segment>
`;

exports[`<FinishGame /> should render a checkmark Icon and the word "Verify" if the current step is 4 2`] = `
<Segment
  basic={true}
>
  <ModalContent>
    <Grid
      columns={2}
    >
      <GridRow>
        <GridColumn
          computer={6}
          mobile={4}
          verticalAlign="middle"
        >
          <StepGroup
            size="mini"
            vertical={true}
          >
            <Step
              active={false}
              completed={true}
            >
              <Icon
                as="i"
                name="info"
                size="small"
              />
              <Responsive
                as={[Function]}
                minWidth={992}
              >
                <StepTitle>
                  Directions
                </StepTitle>
              </Responsive>
            </Step>
            <Step
              active={false}
              completed={true}
            >
              <Icon
                as="i"
                name="hashtag"
                size="small"
              />
              <Responsive
                as={[Function]}
                minWidth={992}
              >
                <StepTitle>
                  Hashtag
                </StepTitle>
              </Responsive>
            </Step>
            <Step
              active={false}
              completed={true}
            >
              <Icon
                as="i"
                name="twitter"
                size="small"
              />
              <Responsive
                as={[Function]}
                minWidth={992}
              >
                <StepTitle>
                  Searching
                </StepTitle>
              </Responsive>
            </Step>
            <Step
              active={false}
              completed={true}
            >
              <Icon
                as="i"
                loading={false}
                name="spinner"
                size="small"
              />
              <Responsive
                as={[Function]}
                minWidth={992}
              >
                <StepTitle>
                  Processing
                </StepTitle>
              </Responsive>
            </Step>
            <Step
              active={true}
              completed={false}
            >
              <Icon
                as="i"
                name="pointing down"
                size="small"
              />
              <Responsive
                as={[Function]}
                minWidth={992}
              >
                <StepTitle>
                  Confirm
                </StepTitle>
              </Responsive>
            </Step>
          </StepGroup>
        </GridColumn>
        <GridColumn
          computer={10}
          mobile={12}
        >
          <RenderStep
            fetchImages={[Function]}
            hashtag="656liv4manure0"
            imageSource=""
            images={null}
            loading={false}
            onResults={[Function]}
            processer={
              TesseractWorker {
                "_currentJob": null,
                "_queue": Array [],
                "create": [Function],
                "version": "1.0.10",
                "worker": null,
                "workerOptions": Object {
                  "langPath": "http://cdn.rawgit.com/naptha/tessdata/gh-pages/3.02/",
                  "workerPath": "/Users/scotthorn/repos/Game-Manager-V3/node_modules/tesseract.js/src/node/worker.js",
                },
              }
            }
            step={4}
          />
        </GridColumn>
      </GridRow>
    </Grid>
  </ModalContent>
  <Segment
    basic={true}
    floated="right"
  >
    <ModalActions>
      <Button
        as="button"
        basic={true}
        color="red"
        inverted={true}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="remove"
        />
         Cancel
      </Button>
      <Button
        as="button"
        color="green"
        disabled={false}
        inverted={true}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="checkmark"
        />
        Verify
      </Button>
    </ModalActions>
  </Segment>
  <Header
    color="teal"
    content="Results"
  />
  <Image
    as="img"
    fluid={true}
    src=""
    ui={true}
  />
</Segment>
`;
